src/lib.rs

```diff
@@ -193,8 +220,8 @@ impl AsyncWrite for SocketWrapper {
     }
 }
 
-enum SocketTypeBuilder<'a, T: ToSocketAddrs + Clone, B: ToOwned<Owned = T> + ?Sized> {
```

Comment:
I agree, I wanted to have the library more ergonomic but in the end these generics just complicate everything

Reply:
I've re-introduced to type bound `ToSocketAddrs`, tell me what you think about it!